src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CryptoHelpers.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Security.Cryptography.Xml
+{
+    internal static class CryptoHelpers
+    {
+        public static object CreateFromName(string name)
+        {
+            switch (name)
```

Comment:
As already discussed, while I like the idea of removing the dependency on `CryptoConfig`, this is a change to the structure of the code which I though is out of scope. Are we happy doing this now or should we port the class first?

Would a `Dictionary<string, Type> uriToType` or `Dictionary<string, Func<SignatureTransform>>` be more appropriate? This allows new algorithms to be added using an equivalent of `AddAlgorithm`.